Show Common-backing language and prevent it from double-counting and displaying separately

DocSchlock opened the pull request:
Closes #13675

GMs can still alter the common-linked language on PC sheets in case it becomes needed.

stwlam commented:
I'm not sure that this is the most intuitive solution, and the actual issue can be addressed just by not counting both Common:tm: and the language that equates to Common:tm: when tallying number of languages known.

DocSchlock commented:
@stwlam 

Changed it to show the backing language as a tooltip instead and reworked the logic to not count the Common-backing language against the total language slots in the sheet.

stwlam commented on src/module/actor/character/sheet.ts:
```diff
@@ -315,25 +315,48 @@ class CharacterSheetPF2e<TActor extends CharacterPF2e> extends CreatureSheetPF2e
             );
-            const isOverMax = languagesBuild.value > languagesBuild.max;
+
+            const deductBackingLanguage = commonLanguage
```
This is (or should be) guaranteed to be present

CarlosFdez replied:
Even if its not, the following logic will still work out fine, the actor wouldn't have a common language of null.

DocSchlock replied:
Bad variable name there. I refactored it - it's supposed to be the backing language.

stwlam commented on src/module/actor/character/sheet.ts:
```diff
@@ -315,23 +315,50 @@ class CharacterSheetPF2e<TActor extends CharacterPF2e> extends CreatureSheetPF2e
+
+            // find if the backing language is present in character's languages
+            // this means characters explictly selected it
+            const deductBackingLanguage = commonBackingLanguage
```
Same

DocSchlock replied:
Cleaned up

stwlam commented on src/module/actor/character/sheet.ts:
```diff
@@ -315,23 +315,50 @@ class CharacterSheetPF2e<TActor extends CharacterPF2e> extends CreatureSheetPF2e
         sheetData.abpEnabled = AutomaticBonusProgression.isEnabled(actor);
 
+        // get the backing language for common
+        const commonBackingLanguage = game.settings.get("pf2e", "homebrew.languageRarities").commonLanguage;
```
Retrieve from cache

DocSchlock replied:
Added to cache and called from there.

stwlam commented:
Organizing this a little differently so that actor data isn't being calculated in a sheet class
